File: iosApp/iosApp/ComponentViews/LineCard.swift
//
//  LineCard.swift
//  iosApp
//
//  Created by Simon, Emma on 6/18/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import shared
import SwiftUI

struct LineCard<Content: View>: View {
    let line: Line
    let routes: [Route]
    let pinned: Bool
    let onPin: (String) -> Void
    let content: () -> Content

    @ScaledMetric private var modeIconHeight: CGFloat = 24

    var body: some View {
        TransitCard(header: {
            LineHeader(line: line, routes: routes) {
                PinButton(pinned: pinned, action: { onPin(line.id) })
            }
        }, content: content)
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/LineCard.kt
package com.mbta.tid.mbta_app.android.component

import androidx.compose.runtime.Composable
import com.mbta.tid.mbta_app.model.Line
import com.mbta.tid.mbta_app.model.Route

@Composable
fun LineCard(
    line: Line,
    routes: List<Route>,
    pinned: Boolean,
    onPin: (String) -> Unit,
    content: @Composable () -> Unit
) {
    TransitCard(
        header = { LineHeader(line, routes) { PinButton(pinned = pinned) { onPin(line.id) } } },
        content
    )
}
